process/block/preprocess/basePreProcess.go

```diff
@@ -71,18 +71,19 @@ func (bpp *basePreProcess) removeDataFromPools(body block.Body, miniBlockPool st
 	return nil
 }
 
-func (bpp *basePreProcess) restoreMiniBlock(miniBlock *block.MiniBlock, miniBlockPool storage.Cacher, restoredHash []byte) error {
+func (bpp *basePreProcess) restoreMiniBlock(miniBlock *block.MiniBlock, miniBlockPool storage.Cacher) ([]byte, error) {
 	miniBlockHash, err := core.CalculateHash(bpp.marshalizer, bpp.hasher, miniBlock)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
+	restoredHash := []byte{}
```

Comment:
It's not idiomatic to initialise with empty literal, we can have nil instead: `var restoredHash []byte`

Reply:
done